docker-compose.yml

```diff
@@ -1,15 +1,56 @@
 services:
+  db:
+    restart: always
+    image: postgres
+    container_name: uzum-intern-postgres-db
+    environment:
+      POSTGRES_USER: your_username
+      POSTGRES_PASSWORD: your_password
+      POSTGRES_DB: uzum_intern_database
+      POSTGRES_PORT: 5432
+    expose:
+      - "5432"
+    ports:
+      - "5432:5432"
+    volumes:
+      - uzum_intern_postgres_data:/var/lib/postgresql/data/
+
+  pgadmin:
+    image: dpage/pgadmin4
+    ports:
+      - 5454:5454/tcp
+    environment:
+      PGADMIN_DEFAULT_EMAIL: admin@gmail.com
+      PGADMIN_DEFAULT_PASSWORD: root
+      PGADMIN_LISTEN_PORT: 5454
+
+
   main-app:
     build: main-app
     ports:
       - 8080:8080
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:postgresql://uzum-intern-postgres-db:5432/uzum_intern_database
+      SPRING_DATASOURCE_USERNAME: your_username
+      SPRING_DATASOURCE_PASSWORD: your_password
 
   notification-app:
     build: notification-app
     ports:
       - 8081:8081
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:postgresql://uzum-intern-postgres-db:5432/uzum_intern_database
+      SPRING_DATASOURCE_USERNAME: your_username
+      SPRING_DATASOURCE_PASSWORD: your_password
 
   payment-app:
     build: payment-app
     ports:
-      - 8082:8082
\ No newline at end of file
+      - 8082:8082
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:postgresql://uzum-intern-postgres-db:5432/uzum_intern_database
```

Comment:
Do you want use shared database for all services ?

Reply:
Yes, single database for all three services.